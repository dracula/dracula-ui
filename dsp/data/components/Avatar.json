{
  "class": "component",
  "type": "page",
  "id": "Avatar",
  "name": "Avatar",
  "last_updated": "2021-03-14T22:07:21.257Z",
  "last_updated_by": "System",
  "description": "The Avatar component is used to represent a user.\n\nIt displays a profile picture, or falls back to the user's initials\nif a profile pic is not provided.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "avatar",
    "languages": {
      "javascriptreact": "<Avatar\n  title=\"Netto Farah\"\n  src=\"https://pbs.twimg.com/profile_images/1217874327137185794/rFqSV_h0_400x400.jpg\"\n/>;\n",
      "html": "<span\n  class=\"drac-avatar drac-bg-green-transparent drac-text-green\"\n  style=\"\n    background-image: url(https://pbs.twimg.com/profile_images/1217874327137185794/rFqSV_h0_400x400.jpg);\n  \"\n></span>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "basic usage",
      "html": "<span\n  class=\"drac-avatar drac-bg-green-transparent drac-text-green\"\n  style=\"\n    background-image: url(https://pbs.twimg.com/profile_images/1217874327137185794/rFqSV_h0_400x400.jpg);\n  \"\n></span>\n",
      "react": "<Avatar\n  title=\"Netto Farah\"\n  src=\"https://pbs.twimg.com/profile_images/1217874327137185794/rFqSV_h0_400x400.jpg\"\n/>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Avatar.png",
      "svgData": "./dsp/assets/svgs/Avatar.svg"
    },
    {
      "title": "border: normal",
      "html": "<span class=\"drac-avatar drac-bg-green-transparent drac-text-green\"\n  ><span color=\"white\" class=\"drac-text drac-text-white\">NF</span></span\n>\n",
      "react": "<Avatar title=\"Netto Farah\" borderVariant=\"normal\" />;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Avatarborder: normal.png",
      "svgData": "./dsp/assets/svgs/Avatarborder: normal.svg"
    },
    {
      "title": "border: large",
      "html": "<span\n  class=\"drac-avatar drac-bg-green-transparent drac-text-green drac-avatar-lg-stroke\"\n  ><span color=\"white\" class=\"drac-text drac-text-white\">NF</span></span\n>\n",
      "react": "<Avatar title=\"Netto Farah\" borderVariant=\"large\" />;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Avatarborder: large.png",
      "svgData": "./dsp/assets/svgs/Avatarborder: large.svg"
    },
    {
      "title": "border: none",
      "html": "<span\n  class=\"drac-avatar drac-bg-green-transparent drac-text-green drac-avatar-no-border\"\n  ><span color=\"white\" class=\"drac-text drac-text-white\">NF</span></span\n>\n",
      "react": "<Avatar title=\"Netto Farah\" borderVariant=\"none\" />;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Avatarborder: none.png",
      "svgData": "./dsp/assets/svgs/Avatarborder: none.svg"
    },
    {
      "title": "Custom variants",
      "html": "<div>\n  <span class=\"drac-avatar drac-bg-green-transparent drac-text-green\"\n    ><span color=\"white\" class=\"drac-text drac-text-white\">NF</span></span\n  ><span\n    class=\"drac-avatar drac-bg-green-transparent drac-text-green drac-avatar-subtle\"\n    ><span color=\"white\" class=\"drac-text drac-text-white\">ZR</span></span\n  >\n</div>\n",
      "react": "<div>\n  <Avatar title=\"Netto Farah\" variant=\"normal\" />\n  <Avatar title=\"Zeno Rocha\" variant=\"subtle\" />\n</div>;\n",
      "docs": "Avatars accept multiple variants, like in the example below...",
      "screenshot": "./dsp/assets/AvatarCustom variants.png",
      "svgData": "./dsp/assets/svgs/AvatarCustom variants.svg"
    },
    {
      "title": "Custom colors",
      "html": "<div>\n  <span\n    class=\"drac-avatar drac-bg-green-transparent drac-text-green\"\n    color=\"black\"\n    ><span color=\"white\" class=\"drac-text drac-text-white\">NF</span></span\n  ><span\n    class=\"drac-avatar drac-bg-green-transparent drac-text-green\"\n    color=\"cyan\"\n    ><span color=\"white\" class=\"drac-text drac-text-white\">NF</span></span\n  >\n</div>\n",
      "react": "<div>\n  <Avatar title=\"Netto Farah\" color=\"black\" />\n  <Avatar title=\"Netto Farah\" color=\"cyan\" />\n</div>;\n",
      "docs": "Avatars accept colors, like in the example below...",
      "screenshot": "./dsp/assets/AvatarCustom colors.png",
      "svgData": "./dsp/assets/svgs/AvatarCustom colors.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "title": {
      "defaultValue": null,
      "description": "The title or name to be used in the avatar.",
      "name": "title",
      "parent": {
        "fileName": "src/components/Avatar/Avatar.tsx",
        "name": "AvatarProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        }
      ],
      "required": true,
      "type": { "name": "string" }
    },
    "themeColor": {
      "defaultValue": null,
      "description": "The theme color for the avatar.\nThis property controls the border and glow colors for the component.",
      "name": "themeColor",
      "parent": {
        "fileName": "src/components/Avatar/Avatar.tsx",
        "name": "AvatarProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        }
      ],
      "required": false,
      "type": {
        "name": "\"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "variant": {
      "defaultValue": null,
      "description": "The variant property allows for consumers to choose from\na regular avatar, or a subtle variation.",
      "name": "variant",
      "parent": {
        "fileName": "src/components/Avatar/Avatar.tsx",
        "name": "AvatarProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        }
      ],
      "required": false,
      "type": { "name": "\"normal\" | \"subtle\" | undefined" }
    },
    "borderVariant": {
      "defaultValue": null,
      "description": "Allows for controlling the border of the avatar.\n`large` -> to use a thicker border.\n`none` -> hides the border.",
      "name": "borderVariant",
      "parent": {
        "fileName": "src/components/Avatar/Avatar.tsx",
        "name": "AvatarProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        }
      ],
      "required": false,
      "type": { "name": "\"normal\" | \"large\" | \"none\" | undefined" }
    },
    "src": {
      "defaultValue": null,
      "description": "Sets the source for the profile picture.",
      "name": "src",
      "parent": {
        "fileName": "src/components/Avatar/Avatar.tsx",
        "name": "AvatarProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        }
      ],
      "required": false,
      "type": { "name": "string | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "The Avatar component is used to represent a user.\n\nIt displays a profile picture, or falls back to the user's initials\nif a profile pic is not provided.",
    "displayName": "Avatar",
    "methods": [],
    "props": {
      "title": {
        "defaultValue": null,
        "description": "The title or name to be used in the avatar.",
        "name": "title",
        "parent": {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Avatar/Avatar.tsx",
            "name": "AvatarProps"
          }
        ],
        "required": true,
        "type": { "name": "string" }
      },
      "themeColor": {
        "defaultValue": null,
        "description": "The theme color for the avatar.\nThis property controls the border and glow colors for the component.",
        "name": "themeColor",
        "parent": {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Avatar/Avatar.tsx",
            "name": "AvatarProps"
          }
        ],
        "required": false,
        "type": {
          "name": "\"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
        }
      },
      "variant": {
        "defaultValue": null,
        "description": "The variant property allows for consumers to choose from\na regular avatar, or a subtle variation.",
        "name": "variant",
        "parent": {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Avatar/Avatar.tsx",
            "name": "AvatarProps"
          }
        ],
        "required": false,
        "type": { "name": "\"normal\" | \"subtle\" | undefined" }
      },
      "borderVariant": {
        "defaultValue": null,
        "description": "Allows for controlling the border of the avatar.\n`large` -> to use a thicker border.\n`none` -> hides the border.",
        "name": "borderVariant",
        "parent": {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Avatar/Avatar.tsx",
            "name": "AvatarProps"
          }
        ],
        "required": false,
        "type": { "name": "\"normal\" | \"large\" | \"none\" | undefined" }
      },
      "src": {
        "defaultValue": null,
        "description": "Sets the source for the profile picture.",
        "name": "src",
        "parent": {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Avatar/Avatar.tsx",
            "name": "AvatarProps"
          }
        ],
        "required": false,
        "type": { "name": "string | undefined" }
      }
    }
  }
}
