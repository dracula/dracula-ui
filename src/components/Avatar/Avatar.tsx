import cx from 'classnames/dedupe'
import { first, last } from 'lodash'
import React, { HTMLAttributes } from 'react'
import { colors as backgroundColors } from '../../base/colors'
import {
  cleanProps,
  MarginMixin,
  marginMixin,
  PaddingMixin,
  paddingMixin
} from '../../base/spacing'
import { Text, textColors } from '../../components/Text/Text'

export const AvatarVariants = {
  normal: 'drac-avatar',
  subtle: 'drac-avatar-subtle'
}

export const AvatarBorderVariants = {
  normal: '',
  large: 'drac-avatar-lg-stroke',
  none: 'drac-avatar-no-border'
}

/** Avatar Props */
export interface AvatarProps
  extends HTMLAttributes<HTMLSpanElement & HTMLImageElement>,
    PaddingMixin,
    MarginMixin {
  /**
   * The title or name to be used in the avatar.
   */
  title: string

  /**
   * The theme color for the avatar.
   * This property controls the border and glow colors for the component.
   */
  themeColor?: keyof typeof backgroundColors

  /**
   * The variant property allows for consumers to choose from
   * a regular avatar, or a subtle variation.
   */
  variant?: keyof typeof AvatarVariants

  /**
   * Allows for controlling the border of the avatar.
   * `large` -> to use a thicker border.
   * `none` -> hides the border.
   */
  borderVariant?: keyof typeof AvatarBorderVariants

  /**
   * Sets the source for the profile picture.
   */
  src?: string
}

/**
 * The Avatar component is used to represent a user.
 *
 * It displays a profile picture, or falls back to the user's initials
 * if a profile pic is not provided.
 */
export const Avatar = (props: AvatarProps) => {
  const { themeColor, variant, borderVariant, src, title, ...htmlProps } = props

  const backgroundClass = `${
    backgroundColors[themeColor ?? 'green']
  }-transparent`

  const classes = cx(
    'drac-avatar',
    props.className,
    backgroundClass,
    textColors[themeColor ?? 'green'],
    AvatarVariants[variant ?? 'normal'],
    AvatarBorderVariants[borderVariant ?? 'normal'],
    ...paddingMixin(props),
    ...marginMixin(props)
  )
  const names = props.title.split(' ')

  const f = first(names)?.slice(0, 1)
  const l = last(names)?.slice(0, 1)

  const style: React.HTMLAttributes<HTMLSpanElement>['style'] = {
    ...props.style
  }

  if (props.src) {
    style.backgroundImage = `url('${props.src}')`
  }

  return (
    <span className={classes} style={style} {...cleanProps(htmlProps)}>
      {!props.src && (
        <Text color={props.themeColor ?? 'white'}>
          {f}
          {l}
        </Text>
      )}
    </span>
  )
}

Avatar.displayName = 'Avatar'
