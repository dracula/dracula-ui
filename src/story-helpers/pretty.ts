import prettier from 'prettier'

export function pretty(
  code: string,
  parser: prettier.Options['parser'],
  options?: prettier.Options
) {
  return prettier.format(code, {
    ...options,
    parser
  })
}
