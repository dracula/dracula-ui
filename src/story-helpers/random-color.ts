import { colors } from '../base/colors'

export const colorname = () => {
  const colorNames = Object.keys(colors)
  const index = Math.round(Math.random() * colorNames.length + 1)
  return colorNames[index]
}
