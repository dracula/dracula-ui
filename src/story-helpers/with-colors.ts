import { ColorMap } from '../base/colors'

export function withColors(
  colorMap: ColorMap,
  fn: (name: string, value: string) => unknown
) {
  return Object.entries(colorMap).map(([name, value]) => {
    return fn(name, value)
  })
}
