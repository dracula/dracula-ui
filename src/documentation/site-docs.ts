import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from './render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from './pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'
import os from 'os'

interface Documentation {
  basic: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder> | SnapshotBuilder
}

jest.setTimeout(10000)

const RECORDING = process.env.RECORDING

export function docs<T>(
  componentClass: React.FunctionComponent<T> | string,
  documentation: Documentation
) {
  const name =
    typeof componentClass === 'string'
      ? componentClass
      : componentClass.displayName ?? componentClass.name

  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeAll(async () => {
      if (!RECORDING) {
        return
      }

      let executablePath = undefined
      if (os.arch() === 'arm64') {
        executablePath = '/Applications/Chromium.app/Contents/MacOS/Chromium'
      }

      browser = await puppeteer.launch({
        headless: true,
        devtools: true,
        executablePath
      })
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      if (!RECORDING) {
        return
      }

      await browser.close()
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basic()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      if (RECORDING) {
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          null,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg
      }

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations: SnapshotBuilder[] = ([] as SnapshotBuilder[]).concat(
      documentation.variations()
    )

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        if (RECORDING) {
          const [screenshot, svg] = await componentScreenshot(
            browser,
            snapshot,
            variation,
            name
          )

          snapshot.screenshot = screenshot
          snapshot.svgData = svg
        }

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
