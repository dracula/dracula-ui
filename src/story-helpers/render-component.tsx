import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { pretty } from './pretty'

Enzyme.configure({ adapter: new Adapter() })

export type ComponentExample = {
  html: string
  docs: string
  title: string
  react?: any
  screenshot?: string
  svgData?: string
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): ComponentExample {
  let component = comp()
  const html = mount(component).html()
  let wrapped = false

  // wrap the component in a div when it's not wrapped already
  // we do this so we can get the actual component example to not be shallow rendered.
  // Shallow rendering a parent "div" gives us a nice JSX representation of the example.
  if (component.type !== 'div') {
    component = <div>{component}</div>
    wrapped = true
  }

  let react = shallow(component).debug({ verbose: true })

  if (wrapped) {
    // replace wrapper div ðŸ˜¬
    react = react
      .replace(/^<div>\n/, '')
      .replace(/\n<\/div>$/, '')
      .trim()
  }

  return {
    title,
    html: pretty(html, 'html'),
    react: pretty(react, 'typescript'),
    docs
  }
}
