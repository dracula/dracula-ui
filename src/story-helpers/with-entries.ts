import { SnapshotBuilder } from '../story-helpers/render-component'

export function withEntries<T>(
  map: T,
  fn: (name: keyof T, value: T[keyof T]) => SnapshotBuilder
) {
  return Object.entries(map).map(([name, value]) => {
    return fn(name as keyof T, value as T[keyof T])
  })
}
