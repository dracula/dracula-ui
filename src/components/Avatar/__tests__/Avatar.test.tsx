import { Avatar, AvatarBorderVariants } from '@/components/Avatar/Avatar'
import {
  buildSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import { siteDocs } from '@/documentation/site-docs'
import { withEntries } from '@/story-helpers/with-entries'
import React from 'react'

siteDocs(Avatar, {
  basicUsage() {
    return buildSnapshot('basic usage', () => (
      <Avatar
        title="Netto Farah"
        src="https://pbs.twimg.com/profile_images/1217874327137185794/rFqSV_h0_400x400.jpg"
      />
    ))
  },
  variations() {
    const snap = buildSnapshot(
      'Custom colors',
      () => (
        <div>
          <Avatar title="Netto Farah" color="black" />
          <Avatar title="Netto Farah" color="cyan" />
        </div>
      ),
      `
    Avatars accept colors, like in the example below...
    `
    )

    const variants = buildSnapshot(
      'Custom variants',
      () => (
        <div>
          <Avatar title="Netto Farah" variant="normal" />
          <Avatar title="Zeno Rocha" variant="subtle" />
        </div>
      ),
      `
    Avatars accept multiple variants, like in the example below...
    `
    )

    const borders = withEntries(AvatarBorderVariants, (variant) => {
      return buildSnapshot(`border: ${variant}`, () => (
        <Avatar title="Netto Farah" borderVariant={variant} />
      ))
    })

    return [...borders, variants, snap] as SnapshotBuilder[]
  }
})
