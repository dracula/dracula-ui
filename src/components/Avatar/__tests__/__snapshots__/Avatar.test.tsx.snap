// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Avatar Basic Usage 1`] = `
Object {
  "docs": "The most basic instance of an Avatar requires a src image and a title to be used for accessibility or as a fallback in case the image can't be loaded.",
  "html": "<span
  class=\\"drac-avatar drac-bg-green-transparent drac-text-green\\"
  style=\\"
    background-image: url(https://pbs.twimg.com/profile_images/1325812776707690496/_iTWBLbr_400x400.jpg);
  \\"
></span>
",
  "react": "<Avatar
  title=\\"Netto Farah\\"
  src=\\"https://pbs.twimg.com/profile_images/1325812776707690496/_iTWBLbr_400x400.jpg\\"
/>;
",
  "title": "Usage",
}
`;

exports[`Site: Avatar Borders 1`] = `
Object {
  "docs": "A border variant allows for customization of the border width of an Avatar, or for an Avatar to be rendered without a border.",
  "html": "<div>
  <span
    class=\\"drac-avatar drac-bg-green-transparent drac-text-green drac-avatar-lg-stroke\\"
    ><span class=\\"drac-text drac-line-height drac-text-white\\">NF</span></span
  ><span
    class=\\"drac-avatar drac-bg-green-transparent drac-text-green drac-avatar-no-border\\"
    ><span class=\\"drac-text drac-line-height drac-text-white\\">NF</span></span
  >
</div>
",
  "react": "<div>
  <Avatar title=\\"Netto Farah\\" borderVariant=\\"large\\" />
  <Avatar title=\\"Netto Farah\\" borderVariant=\\"none\\" />
</div>;
",
  "title": "Borders",
}
`;

exports[`Site: Avatar Colors 1`] = `
Object {
  "docs": "The Avatar component accepts a *color* param that allows for customizing the background and accent color for an Avatar.",
  "html": "<div>
  <span
    class=\\"drac-avatar drac-bg-green-transparent drac-text-green\\"
    color=\\"cyan\\"
    ><span class=\\"drac-text drac-line-height drac-text-white\\">NF</span></span
  >
</div>
",
  "react": "<div>
  <Avatar title=\\"Netto Farah\\" color=\\"cyan\\" />
</div>;
",
  "title": "Colors",
}
`;

exports[`Site: Avatar Variants 1`] = `
Object {
  "docs": "Avatars accept multiple variants, which can be used to render a more subtle variation of the Avatar component.",
  "html": "<div>
  <span
    class=\\"drac-avatar drac-bg-green-transparent drac-text-green drac-avatar-subtle\\"
    ><span class=\\"drac-text drac-line-height drac-text-white\\">ZR</span></span
  ><span class=\\"drac-avatar drac-bg-green-transparent drac-text-green\\"
    ><span class=\\"drac-text drac-line-height drac-text-white\\">NF</span></span
  >
</div>
",
  "react": "<div>
  <Avatar title=\\"Zeno Rocha\\" variant=\\"subtle\\" />
  <Avatar title=\\"Netto Farah\\" variant=\\"normal\\" />
</div>;
",
  "title": "Variants",
}
`;
